src/qengine/gates.cpp

```diff
@@ -37,21 +37,7 @@ void QEngineCPU::CCNOT(bitLenInt control1, bitLenInt control2, bitLenInt target)
         throw std::invalid_argument("CCNOT control bits cannot also be target.");
     }
 
-    const Complex16 pauliX[4] = {
-            Complex16(0.0, 0.0), Complex16(1.0, 0.0),
-            Complex16(1.0, 0.0), Complex16(0.0, 0.0) };
-
-    bitCapInt qPowers[4];
-    bitCapInt qPowersSorted[3];
-    qPowers[1] = 1 << control1;
-    qPowersSorted[0] = qPowers[1];
-    qPowers[2] = 1 << control2;
-    qPowersSorted[1] = qPowers[2];
-    qPowers[3] = 1 << target;
-    qPowersSorted[2] = qPowers[3];
-    qPowers[0] = qPowers[1] + qPowers[2] + qPowers[3];
-    std::sort(qPowersSorted, qPowersSorted + 3);
-    Apply2x2(qPowers[0], qPowers[1] + qPowers[2], pauliX, 3, qPowersSorted, false, false);
```

Comment:
Should there be in qinterface?

Reply:
I see your point, but I think not. It makes more sense if QInterface just contains overload-able bitwise parallel loops and pure virtual single bit prototypes. If we add single bit pass-throughs to the length=1 cases of register methods, the logic becomes circular, even though we expect to overload one or both methods from the interface.